### PR TITLE
update rrtmgp

### LINKED ISSUE #3219: ClimaAtmos should not use data defined in the tests for RRTGMP
https://github.com/CliMA/ClimaAtmos.jl/blob/ab06398bb9ce263a58f8be0b0f484c8235d85ae3/src/parameterized_tendencies/radiation/radiation_utilities.jl#L9-L36

If ClimaAtmos needs this data, RRTGMP should provide a natural way to access it. 

@sriharshakandala can you please fix this?

### PR DESCRIPTION
**Pull request by szy21**
Closes #3219 
Closes #3265 
Closes #3283 

## Content

----
- [ ] I have read and checked the items on the review checklist.

### CODE REVIEW

File: src/parameterized_tendencies/radiation/RRTMGPInterface.jl [hunk 1]
```diff
@@ -805,17 +805,44 @@ function RRTMGPModel(
...
+            bcpo_aero_mass = view(aero_mass, 5, :, :)
+            ocpi_aero_mass = view(aero_mass, 6, :, :)
+            ocpo_aero_mass = view(aero_mass, 7, :, :)
+            set_and_save!(dust_aero_mass, "center_dust_column_mass_density", t..., dict)
+            set_and_save!(ss_aero_mass, "center_ss_column_mass_density", t..., dict)
+            set_and_save!(sulfate_aero_mass, "center_sulfate_column_mass_density", t..., dict)
+            set_and_save!(bcpi_aero_mass, "center_bcpi_column_mass_density", t..., dict)
+            set_and_save!(bcpo_aero_mass, "center_bcpo_column_mass_density", t..., dict)
+            set_and_save!(ocpi_aero_mass, "center_ocpi_column_mass_density", t..., dict)
+            set_and_save!(ocpo_aero_mass, "center_ocpo_column_mass_density", t..., dict)
```
Comment (Sbozzolo) [inline src/parameterized_tendencies/radiation/RRTMGPInterface.jl:843]:
```suggestion
aero_size = DA{FT}(undef, n_aerosols, nlay, ncol)
aero_mass = DA{FT}(undef, n_aerosols, nlay, ncol)

aerosol_names = [
    "dust", "ss", "sulfate", "bcpi", "bcpo", "ocpi", "ocpo"
]

for (i, name) in enumerate(aerosol_names)
    set_and_save!(view(aero_size, i, :, :), "center_$(name)_radius", t..., dict)
    set_and_save!(view(aero_mass, i, :, :), "center_$(name)_column_mass_density", t..., dict)
end
```

Comment (Sbozzolo) [reply src/parameterized_tendencies/radiation/RRTMGPInterface.jl:843]: I think

Comment (szy21) [reply src/parameterized_tendencies/radiation/RRTMGPInterface.jl:843]: Thanks! I was going to test how to make it simpler.

File: src/callbacks/callbacks.jl [hunk 2]
```diff
@@ -168,15 +168,34 @@ NVTX.@annotate function rrtmgp_model_callback!(integrator)
...
+            ]
+
+            for (aerosol_name, prescribed_aerosol_name) in aerosol_names_pair
+                ᶜaero_conc = Fields.array2field(
+                    getproperty(rrtmgp_model, aerosol_name),
+                    axes(Y.c),
+                )
+                if prescribed_aerosol_name in
+                   propertynames(p.tracers.prescribed_aerosols_field)
+                    @. ᶜaero_conc =
```
Comment (szy21) [inline src/callbacks/callbacks.jl:189]: This doesn't work on GPU: https://buildkite.com/clima/climaatmos-ci/builds/20263#01919159-513a-4f3e-bb65-44ec284ec892/205-300. @Sbozzolo @dennisYatunin Is there a way to make it work on GPU (while I can still understand the code...)?

File: src/callbacks/callbacks.jl [hunk 3]
```diff
@@ -168,15 +168,34 @@ NVTX.@annotate function rrtmgp_model_callback!(integrator)
...
+                        getproperty(
+                            p.tracers.prescribed_aerosols_field,
+                            prescribed_aerosol_name,
+                        ) * ᶜΔz
```
Comment (Sbozzolo) [inline src/callbacks/callbacks.jl:193]:
```suggestion
                   propertynames(p.tracers.prescribed_aerosols_field)
                   aerosol_field = getproperty(
                            p.tracers.prescribed_aerosols_field,
                            prescribed_aerosol_name,
                        )
                   @. ᶜaero_conc = aerosol_field * ᶜΔz
```
is my first guess.

Comment (szy21) [reply src/callbacks/callbacks.jl]: it runs!

File: config/model_configs/sphere_aquaplanet_rhoe_equilmoist_allsky_gw_res.yml [hunk 4]
```diff
@@ -2,7 +2,7 @@ z_elem: 25
 z_max: 55000.0
 dz_bottom: 300.0
 dt: "400secs"
-t_end: "1days"
+t_end: "1hours"
```
Comment (szy21) [inline config/model_configs/sphere_aquaplanet_rhoe_equilmoist_allsky_gw_res.yml:5]: I'll revert this.

**Comment by szy21** [comment]
I'll do more test and release rrtmgp before this, but @Sbozzolo Could you take a look at the code change?

File: src/parameterized_tendencies/radiation/radiation_utilities.jl [hunk 5]
```diff
@@ -30,7 +14,7 @@ Loads data from an `NCDataset` from the `RRTMGP.jl` artifact stored in
 dataset.
 """
 function rrtmgp_data_loader(fn, file_name)
-    NC.Dataset(rrtmgp_artifact(file_name), "r") do ds
+    NC.Dataset(file_name, "r") do ds
         fn(ds)
     end
 end
```
Comment (szy21) [inline src/parameterized_tendencies/radiation/radiation_utilities.jl:20]: Do we still need this function?

Comment (Sbozzolo) [reply src/parameterized_tendencies/radiation/radiation_utilities.jl:20]: The function should be replaced with what was introduced here: https://github.com/CliMA/RRTMGP.jl/pull/520

Comment (Sbozzolo) [reply src/parameterized_tendencies/radiation/radiation_utilities.jl:20]: (Which you did)

**Review by Sbozzolo — APPROVED** [review]

**Comment by Sbozzolo** [comment]
Please, remove Pkg from the dependencies before merging this.

**Pull request merged**